app/lib/features/chat/providers/chat_providers.dart

```diff
@@ -106,3 +107,9 @@ final chatMentionsProvider =
   }
   return mentionRecords;
 });
+
+// state of typing notice setting
+final chatTypingEventStateProvider = StateNotifierProvider.autoDispose<
+    ChatTypingEventStateNotifier, TypingEvent?>(
+  (ref) => ChatTypingEventStateNotifier(ref),
+);
```

Comment:
Looking at it after some understanding, can't the entire notifier be slimmed down to a stream provider?

```suggestion
// state of typing notice setting
final chatTypingEventStateProvider = StreamProvider<TypingEvent?>((ref) async* {
  // Connect to an API using sockets, and decode the output
      final userAppSetttings = await ref.watch(userAppSettingsProvider.future);
    if (!userAppSetttings.typingNotice()) {
       return null;
    }
  await for (final event in client.typingEventRx()!) {
    yield event;
  }
});
```

Reply:
I'll try it this way. The initial non-included similar version has another provider aka ```StreamProvider``` which was observable for ```StateNotifier``` but has a problem of still being subscribed to receive events...

Reply:
@gtalha07 minor update to deal with `null` correctly.